### PR TITLE
Updated to 1.15.0

### PR DESCRIPTION
**Pull request by npanpaliya**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

**Comment by JanuszL** [comment]
`Updated to 0.14.0` -> `Updated to 1.14.0`

**Comment by JanuszL** [comment]
The original error seems to be a false positive according to @mzient.

**Comment by npanpaliya** [comment]
> The original error seems to be a false positive according to @mzient.

There are many such errors which are now warnings as per the flag set.

File: recipe/0303-Fixed-errors-encountered-with-GCC-11.patch [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+diff --git a/dali/pipeline/executor/executor_test.cc b/dali/pipeline/executor/executor_test.cc
+index d914054f..2c7aaaeb 100644
+--- a/dali/pipeline/executor/executor_test.cc
++++ b/dali/pipeline/executor/executor_test.cc
+@@ -365,24 +365,24 @@ TYPED_TEST(ExecutorTest, DISABLED_TestDataSetup) {
+     ASSERT_EQ(hws.NumInput(), 0);
+     ASSERT_EQ(hws.NumOutput(), 1);
+     ASSERT_EQ(hws.GetRequestedBatchSize(0), this->batch_size_);
+-    ASSERT_TRUE(hws.OutputIsType<CPUBackend>(0));
++    //ASSERT_TRUE(hws.OutputIsType<CPUBackend>(0));
```
Comment (npanpaliya) [inline recipe/0303-Fixed-errors-encountered-with-GCC-11.patch:21]: @JanuszL - I'd to comment these lines from executor_test as these lines were giving error - `"error: 'this' pointer is null" Wnon-null`

Comment (JanuszL) [reply recipe/0303-Fixed-errors-encountered-with-GCC-11.patch:21]: Interesting. If that helps you feel free. We will try to look into it in spare time but I cannot promise any ETA.

Comment (npanpaliya) [reply recipe/0303-Fixed-errors-encountered-with-GCC-11.patch:21]: Sure @JanuszL . No problem. Since it's a test, we are safe to have this commented.

**Comment by npanpaliya** [comment]
@JanuszL - On ppc64le, this PR gave me following error -
```
/opt/conda/conda-bld/dali_1655480058413/work/dali/kernels/imgproc/structure/connected_components.h:173:18: error: '((const long int*)((char*)&prev_out + offsetof(dali::kernels::connected_components::detail::TensorSlice<unsigned int, 2>,dali::kernels::connected_components::detail::TensorSlice<unsigned int, 2>::size.dali::vec<2, long int>::<unnamed>.dali::vec_base<2, long int>::<unnamed>)))[1]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  173 |       MergeSlices(label_base, prev_out, out_slice, prev_in, in_slice);
      |       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

After looking into the relevant code here https://github.com/NVIDIA/DALI/blob/v1.14.0/dali/kernels/imgproc/structure/connected_components.h#L173, I think for the 1st iteration, `prev_in` and `prev_out` will be uninitialized and still be read in MergeSlices  function. Please let me know if my understanding is correct and how can we initialize TensorSlice prev_in and prev_out at first iteration?

**Comment by npanpaliya** [comment]
> @JanuszL - On ppc64le, this PR gave me following error -
> 
> ```
> /opt/conda/conda-bld/dali_1655480058413/work/dali/kernels/imgproc/structure/connected_components.h:173:18: error: '((const long int*)((char*)&prev_out + offsetof(dali::kernels::connected_components::detail::TensorSlice<unsigned int, 2>,dali::kernels::connected_components::detail::TensorSlice<unsigned int, 2>::size.dali::vec<2, long int>::<unnamed>.dali::vec_base<2, long int>::<unnamed>)))[1]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
>   173 |       MergeSlices(label_base, prev_out, out_slice, prev_in, in_slice);
>       |       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```
> 
> After looking into the relevant code here https://github.com/NVIDIA/DALI/blob/v1.14.0/dali/kernels/imgproc/structure/connected_components.h#L173, I think for the 1st iteration, `prev_in` and `prev_out` will be uninitialized and still be read in MergeSlices function. Please let me know if my understanding is correct and how can we initialize TensorSlice prev_in and prev_out at first iteration?

I've made this change in the code which seems to be working - 
```
diff --git a/dali/kernels/imgproc/structure/connected_components.h b/dali/kernels/imgproc/structure/connected_components.h
index 739a995d..87ad0c7c 100644
--- a/dali/kernels/imgproc/structure/connected_components.h
+++ b/dali/kernels/imgproc/structure/connected_components.h
@@ -167,13 +167,13 @@ enable_if_t<(ndim > 1)> LabelSlice(OutLabel *label_base,
   for (int64_t i = 0; i < n; i++) {
     auto out_slice = out.slice(i);
     auto in_slice = in.slice(i);
+    prev_out = out_slice;
+    prev_in  = in_slice;
     LabelSlice(label_base, out_slice, in_slice, background, engine);

     if (i > 0)
       MergeSlices(label_base, prev_out, out_slice, prev_in, in_slice);

-    prev_out = out_slice;
-    prev_in  = in_slice;
   }
 }
```
Please let me know if you think this fix may have any side effect on the functionality.

**Comment by mzient** [comment]
Hello @npanpaliya,
The fix you propose actually breaks the code The variables are initialized, which can be easily traced through the. Your change makes the previous values equal to the current ones, rendering the "prev" useless.
The variables are not used without initialization - they are initialized in iteration `i == 0` and used only when `i > 0`. Please disable `-Werror` in your compilation and don't try to fix what is not broken. If, for some reason, you cannot disable `-Werror`, then do the following in lines 165-166:
```C++
  TensorSlice<OutLabel, ndim-1> prev_out = {};
  TensorSlice<const InLabel, ndim-1> prev_in = {};
```

**Comment by npanpaliya** [comment]
> Hello @npanpaliya, The fix you propose actually breaks the code The variables are initialized, which can be easily traced through the code. Your change makes the previous values equal to the current ones, rendering the "prev" useless. The variables are not used without initialization - they are initialized in iteration `i == 0` and used only when `i > 0`. Please disable `-Werror` in your compilation and don't try to fix what is not broken. If, for some reason, you cannot disable `-Werror`, then do the following in lines 165-166:
> 
> ```c++
>   TensorSlice<OutLabel, ndim-1> prev_out = {};
>   TensorSlice<const InLabel, ndim-1> prev_in = {};
> ```

Okay. Sure. Thanks @mzient for your suggestion. I'll update my PR by disabling the error.

**Review by cdeepali — APPROVED** [review]

**Pull request merged**